### DISCUSSION TITLE
Best approch to save data

### DISCUSSION THREAD

**Discussion by renrom** [Budibase/budibase#10047] [Help] [answered]
Hello, sorry if my question is a rather basic question :)

I use an external MySQL datasource with a bunch of custom queries for selecting data. 

Now if i am in a edit form i use the 'save row' action when i want to save the data.  Is this the appropiate way? Or is it better to also use a custom 'update' query for this?  

For all the forms i use the save row action, but i tend to move to using a custom query for this updates. Or does it not matter?

**Comment by gmentzik** [comment, marked as answer]
From my experience so far.
If you have the internal budibase row _id available, ie. fetched data using table provider, then you can edit row with _id. IF you only have mysql id or other to match  you need custom queries.
